design/vision.md

```diff
@@ -0,0 +1,21 @@
+This file describes the goals, non-goals and guiding principles of the dotty
+project. It is inspired by the analogous [Neovim vision page][neovim] and is
+specially set up to be able to be discussed and designed entirely within a
+Github PR.
+
+[neovim]: https://neovim.io/charter/
+
+# Project Goals
+* Provide a powerful manager for a single user's dotfiles.
+* Modularise configuration and setup scripting.
+* Allow the users to share their dotfiles as packages.
+* Enable users to harness packages from their peers.
+
+# Project Non-Goals
+* Become a package manager.
+* Suggestion: Manage plugins of any other applications.
```

Comment:
How do we draw the line between setup scripting and managing plugins for other applications?